diff --git a/configure.ac b/configure.ac
index 5f585963bdd5..137861722eb4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6077,6 +6077,7 @@ if test "$cross_compiling" = "yes"; then
     # those expanded by the AC_CONFIG_FILES() call near the end of this configure.ac. For clarity,
     # keep them in the same order as there.
     (cd $SRC_ROOT && tar cf - \
+        configure \
         config.guess \
         bin/get_config_variables \
         solenv/bin/getcompver.awk \
@@ -6100,7 +6101,6 @@ if test "$cross_compiling" = "yes"; then
         extensions/source/macosx/quicklookthumbnail/appex/Info.plist.in \
         ) \
     | (cd CONF-FOR-BUILD && tar xf -)
-    cp configure CONF-FOR-BUILD
     test -d config_build && cp -p config_build/*.h CONF-FOR-BUILD/config_host 2>/dev/null
     (
     unset COM USING_X11 OS CPUNAME
