diff --git a/configure.ac b/configure.ac
index 5f585963bdd5..c48428f54b40 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5364,9 +5364,6 @@ ios*)

     case "$host_cpu" in
     aarch64|arm64)
-        if test "$enable_ios_simulator" = "yes"; then
-            AC_MSG_ERROR([iOS simulator is only available in macOS not iOS])
-        fi
         ;;
     *)
         AC_MSG_ERROR([Unsupported host_cpu $host_cpu for host_os $host_os])
