diff --git a/solenv/gbuild/platform/unxgcc.mk b/solenv/gbuild/platform/unxgcc.mk
index f707ca068494..68725c103c66 100644
--- a/solenv/gbuild/platform/unxgcc.mk
+++ b/solenv/gbuild/platform/unxgcc.mk
@@ -237,10 +237,15 @@ gb_Library_DEFS :=
 gb_Library_SYSPRE := lib
 gb_Library_UNOVERPRE := $(gb_Library_SYSPRE)uno_
 
+ifeq ($(OS_FOR_BUILD),MACOSX)
+gb_Library_PLAINEXT_FOR_BUILD := .dylib
+else
+gb_Library_PLAINEXT_FOR_BUILD := .so
+endif
+
 ifeq ($(DISABLE_DYNLOADING),TRUE)
 
 gb_Library_PLAINEXT := .a
-gb_Library_PLAINEXT_FOR_BUILD := .so
 gb_Library_DLLEXT := .a
 
 else
@@ -248,7 +253,6 @@ else
 gb_Library_TARGETTYPEFLAGS := -shared -Wl,-z,noexecstack
 gb_Library_UDK_MAJORVER := 3
 gb_Library_PLAINEXT := .so
-gb_Library_PLAINEXT_FOR_BUILD := .so
 gb_Library_DLLEXT := .so
 
 endif
