diff --git a/external/argon2/0003-use-make-AR-variable.patch b/external/argon2/0003-use-make-AR-variable.patch
new file mode 100644
index 000000000000..5166efa3e039
--- /dev/null
+++ b/external/argon2/0003-use-make-AR-variable.patch
@@ -0,0 +1,11 @@
+--- a/Makefile
++++ b/Makefile
+@@ -182,7 +182,7 @@ $(LIB_SH): 	$(SRC)
+ 		$(CC) $(CFLAGS) $(LIB_CFLAGS) $(LDFLAGS) $(SO_LDFLAGS) $^ -o $@
+
+ $(LIB_ST): 	$(OBJ)
+-		ar rcs $@ $^
++		$(AR) rcs $@ $^
+
+ .PHONY: clean
+ clean:
diff --git a/external/argon2/UnpackedTarball_argon2.mk b/external/argon2/UnpackedTarball_argon2.mk
index c795a1562974..fc60c8d577e1 100644
--- a/external/argon2/UnpackedTarball_argon2.mk
+++ b/external/argon2/UnpackedTarball_argon2.mk
@@ -16,6 +16,7 @@ $(eval $(call gb_UnpackedTarball_set_patchlevel,argon2,1))
 $(eval $(call gb_UnpackedTarball_add_patches,argon2,\
 	external/argon2/0001-Fix-possible-compiler-error-due-to-undefined-_MSC_VE.patch \
 	$(if $(filter WNT_AARCH64,$(OS)_$(CPUNAME)),external/argon2/0002-Add-WinARM64-vcxproj-config.patch) \
+	external/argon2/0003-use-make-AR-variable.patch \
 	external/argon2/private-symbols.patch.0 \
 ))
 
