diff --git a/desktop/source/lib/init.cxx b/desktop/source/lib/init.cxx
index 2469d7686ed2..4be735867642 100644
--- a/desktop/source/lib/init.cxx
+++ b/desktop/source/lib/init.cxx
@@ -8582,14 +8582,19 @@ static int lo_initialize(LibreOfficeKit* pThis, const char* pAppPath, const char
     else
     {
 #if defined ANDROID || defined EMSCRIPTEN
-        aAppPath = OUString::fromUtf8(lo_get_app_data_dir()) + "/program";
-#else
-        // Fun conversion dance back and forth between URLs and system paths...
-        OUString aAppURL;
-        (void)::osl::Module::getUrlFromAddress(reinterpret_cast<oslGenericFunction>(lo_initialize),
-                                               aAppURL);
-        osl::FileBase::getSystemPathFromFileURL( aAppURL, aAppPath );
+        if (const char* pAppDataDir = lo_get_app_data_dir())
+        {
+            aAppPath = OUString::fromUtf8(pAppDataDir) + "/program";
+        }
+        else
 #endif
+        {
+            // Fun conversion dance back and forth between URLs and system paths...
+            OUString aAppURL;
+            (void)::osl::Module::getUrlFromAddress(reinterpret_cast<oslGenericFunction>(lo_initialize),
+                                                   aAppURL);
+            osl::FileBase::getSystemPathFromFileURL( aAppURL, aAppPath );
+        }

 #ifdef IOS
         // The above gives something like
