diff --git a/solenv/bin/native-code.py b/solenv/bin/native-code.py
index 359598df81a5..f1e5a2f16f42 100755
--- a/solenv/bin/native-code.py
+++ b/solenv/bin/native-code.py
@@ -158,6 +158,8 @@ core_constructor_list = [
     "filter_PDFExportInteractionHandler_get_implementation",
     "filter_PDFFilter_get_implementation",
     "filter_PDFDialog_get_implementation",
+# filter/source/graphic/graphicfilter.component
+    "filter_GraphicExportFilter_get_implementation",
 # filter/source/xmlfilterdetect/xmlfd.component
     "filter_XMLFilterDetect_get_implementation",
 # filter/source/xmlfilteradaptor/xmlfa.component
