diff --git a/external/nss/ExternalProject_nss.mk b/external/nss/ExternalProject_nss.mk
index cd198051c017..0704386ed439 100644
--- a/external/nss/ExternalProject_nss.mk
+++ b/external/nss/ExternalProject_nss.mk
@@ -55,7 +55,7 @@ $(call gb_ExternalProject_get_state_target,nss,build): \
 		$(SRCDIR)/external/nss/nsinstall.py
 	$(call gb_Trace_StartRange,nss,EXTERNAL)
 	$(call gb_ExternalProject_run,build,\
-		$(if $(filter ANDROID FREEBSD LINUX MACOSX,$(OS)),$(if $(filter X86_64,$(CPUNAME)),USE_64=1 CPU_ARCH=x86_64)) \
+		$(if $(filter ANDROID FREEBSD LINUX MACOSX iOS,$(OS)),$(if $(filter X86_64,$(CPUNAME)),USE_64=1 CPU_ARCH=x86_64)) \
 		$(if $(filter AARCH64,$(CPUNAME)),USE_64=1 CPU_ARCH=aarch64) \
 		$(if $(filter POWERPC64,$(CPUNAME)),USE_64=1 CPU_ARCH=ppc64le) \
 		$(if $(filter MACOSX,$(OS)),\
