diff --git a/solenv/gbuild/UnpackedTarball.mk b/solenv/gbuild/UnpackedTarball.mk
--- a/solenv/gbuild/UnpackedTarball.mk
+++ b/solenv/gbuild/UnpackedTarball.mk
@@ -143,7 +143,7 @@ $(call gb_Helper_abbreviate_dirs,\
 			for p in $(UNPACKED_PATCHES); do \
 				pl=$(UNPACKED_PATCHLEVEL); \
 				s=$${p##*.}; case "$$s" in [0-9]$(CLOSE_PAREN) pl="$$s"; ;; esac ; \
-				$(GNUPATCH) $(UNPACKED_PATCHFLAGS) -f -s "-p$$pl" --fuzz=0 < "$$p"; \
+				"$(GNUPATCH)" $(UNPACKED_PATCHFLAGS) -f -s "-p$$pl" --fuzz=0 < "$$p"; \
 				if test "$$?" -ne 0; then echo "Patch FAILED: $$p"; exit 1; fi;\
 			done && \
 		) \
