diff --git a/sw/source/core/access/accpara.hxx b/sw/source/core/access/accpara.hxx
index 357c4f17c77d..a2f2744215e7 100644
--- a/sw/source/core/access/accpara.hxx
+++ b/sw/source/core/access/accpara.hxx
@@ -234,7 +234,10 @@ public:
     SwAccessibleParagraph( std::shared_ptr<SwAccessibleMap> const& pInitMap,
                            const SwTextFrame& rTextFrame );
 
-    inline operator css::accessibility::XAccessibleText *();
+    operator css::accessibility::XAccessibleText *()
+    {
+        return static_cast< css::accessibility::XAccessibleEditableText * >( this );
+    }
 
     virtual bool HasCursor() override;   // required by map to remember that object
 
@@ -392,11 +395,6 @@ public:
     virtual css::uno::Sequence< css::beans::PropertyValue > SAL_CALL getRunAttributes( sal_Int32 nIndex, const css::uno::Sequence< OUString >& aRequestedAttributes ) override;
 };
 
-inline SwAccessibleParagraph::operator css::accessibility::XAccessibleText *()
-{
-    return static_cast< css::accessibility::XAccessibleEditableText * >( this );
-}
-
 #endif
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
